Entities/ACrab.cpp

```diff
@@ -2152,18 +2152,23 @@ void ACrab::Update()
 
     if (m_pJetpack && m_pJetpack->IsAttached())
     {
-        // Start Jetpack burn
-        if (m_Controller.IsState(BODY_JUMPSTART) && m_JetTimeLeft > 0)
-        {
-            m_pJetpack->TriggerBurst();
-            // This is to make sure se get loose from being stuck
-            m_ForceDeepCheck = true;
-            m_pJetpack->EnableEmission(true);
-            // Quadruple this for the burst
-            m_JetTimeLeft -= g_TimerMan.GetDeltaTimeMS() * 10;
-            if (m_JetTimeLeft < 0)
-                m_JetTimeLeft = 0;
-        }
+		if (m_JetTimeTotal > 0) {
+			// Throttle depletes based on throttle range
+			float jetTimeRatio = std::max(m_JetTimeLeft / m_JetTimeTotal, 0.0F);
+			m_pJetpack->SetThrottle(jetTimeRatio * 2.0F - 1.0F);
```

Comment:
Not your fault but I feel like AEmitter throttle math is needlessly complicated. Maybe I'm wrong and it's actually a good solution, but not knowing a lot about it it seems messy as hell lol.

Reply:
I assure you it's only weird for the first few seconds, but in the end I can't really think of a less messy way to do it :p

I think this is how throttle is handled in most games (1 = max obverse, -1 = max reverse).

Reply:
Hmmm yeah, I think you're right, I think my sleepy brain wasn't understanding what throttle really meant. I.e. positive throttle speeds up your emission rate and negative throttle slows it down, between 0 - 200% based on the min/max throttle ranges. I think those variables (m_MinThrottleRange and m_MaxThrottleRange) need a rename though, seems to me they define the multiplier for negative/positive throttle, which also makes them inherently the upper/lower limit for this.
Maybe throttle limits and throttle multipliers would be less confusing if they were split into 2 variables, though I'm not sure that's a good idea.

Reply:
Nobody's used them for anything until now anyway. Like I said, it's gonna be confusing for the first few seconds until you see what it does.

Reply:
Yeah fair. I agree the system makes sense now that I know what's going on, but I think a variable rename for these 2 is definitely needed (probably in another branch), because the current names are really bad imo.

Reply:
Done in #321. Feel free to grab that if you want, or leave it for someone else, and resolve this comment.